### PR TITLE
Pattern API: Add `curation` parameter 

### PR DESCRIPTION
**Pull request by ryelle**
Add a new `curation` parameter to the API, which can be one of `all`, `core`, or `community`. It defaults to `all` in the API so that requests continue to work as expected.

See #580.

This parameter is also supported by the api.w.org endpoint, which will be used by the pattern explorer some time in the future (see https://github.com/WordPress/gutenberg/issues/44611)

### How to test the changes in this Pull Request:

This is an example request that will show "community" patterns.

`http://localhost:8888/wp-json/wp/v2/wporg-pattern?locale=en_US&curation=community`

Try the request with different `curation` values:

- `core` should return only patterns with the `core` keyword
- `community` should return only patterns without the `core` keyword (check `keyword_slugs` if you're not sure)
- `all` should return all patterns, no filtering
- any other value should trigger a validation error

You can also test on a sandbox, `https://api.wordpress.org/patterns/1.0/?locale=en_US&curation=core`

### CODE REVIEW

**Review by outdoor2kode — APPROVED** [review]
Tested on a sandbox and LGTM 👍

File: public_html/wp-content/plugins/pattern-directory/includes/pattern-post-type.php [hunk 1]
```diff
@@ -716,6 +728,28 @@ function filter_patterns_rest_query( $args, $request ) {
...
+	// If `curation` is passed and either `core` or `community`, we should
+	// filter the result. If `curation=all`, no filtering is needed.
+	if ( isset( $request['curation'] ) ) {
+		if ( 'core' === $request['curation'] ) {
+			// Patterns with the core keyword.
+			$args['tax_query']['core_keyword'] = array(
+				'taxonomy' => 'wporg-pattern-keyword',
+				'field'    => 'slug',
+				'terms'    => 'core',
+				'operator' => 'IN',
```
Comment (StevenDufresne) [inline public_html/wp-content/plugins/pattern-directory/includes/pattern-post-type.php:740]: This looks like the only difference between the queries? If so, is this worth simplifying?

Comment (ryelle) [reply public_html/wp-content/plugins/pattern-directory/includes/pattern-post-type.php:740]: The operator is the only difference, but I like that this keeps the full sub-query together.

Comment (StevenDufresne) [reply public_html/wp-content/plugins/pattern-directory/includes/pattern-post-type.php:740]: Yep. 👍

**Pull request merged**